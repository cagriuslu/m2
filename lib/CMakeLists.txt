add_subdirectory(box2d)
add_subdirectory(expected)
add_subdirectory(googletest)
add_subdirectory(protobuf)
add_subdirectory(SDL2)

add_subdirectory(z)
add_subdirectory(png)
add_subdirectory(SDL2_image)

add_subdirectory(freetype)
add_subdirectory(SDL2_ttf)

# Define the macro that will eventually build some of these dependencies
macro(BuildDependency dependency)
    # This macro uses execute_process to run cmake again because it needs to finish a successful run to generate the
    # build script. BUILDING_DEPS is set to ON to signal to the executed cmake process that the configuration should
    # return successfully, albeit prematurely.
    if(BUILDING_DEPS)
        return()
    else()
        message("Will build ${dependency}...")
        execute_process(
                COMMAND "${CMAKE_COMMAND}" -G ${CMAKE_GENERATOR} -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} -DBUILDING_DEPS=ON -B ${CMAKE_BINARY_DIR} -S ${CMAKE_SOURCE_DIR}
                OUTPUT_FILE ${CMAKE_BINARY_DIR}/${dependency}-cmake.out
                ERROR_FILE ${CMAKE_BINARY_DIR}/${dependency}-cmake.err
                RESULT_VARIABLE EXIT_CODE)
        if(NOT EXIT_CODE EQUAL "0")
            message(FATAL_ERROR "Failed to configure ${dependency}")
        endif()
        execute_process(
                COMMAND "${CMAKE_COMMAND}" --build ${CMAKE_BINARY_DIR} --target ${dependency}
                OUTPUT_FILE ${CMAKE_BINARY_DIR}/${dependency}-build.out
                ERROR_FILE ${CMAKE_BINARY_DIR}/${dependency}-build.err
                RESULT_VARIABLE EXIT_CODE)
        if(NOT EXIT_CODE EQUAL "0")
            message(FATAL_ERROR "Failed to build ${dependency}")
        endif()
    endif()
endmacro()
