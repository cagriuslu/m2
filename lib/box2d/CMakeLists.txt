include(FetchContent)
FetchContent_Declare(
    libbox2d
    URL ${CMAKE_CURRENT_LIST_DIR}/box2d-2.4.1.zip
    URL_HASH SHA256=0cb512dfa5be79ca227cd881b279adee61249c85c8b51caf5aa036b71e943002
)
set(BOX2D_BUILD_UNIT_TESTS OFF CACHE INTERNAL "Disable box2d unit tests")
set(BOX2D_BUILD_TESTBED OFF CACHE INTERNAL "Disable box2d testbed")
