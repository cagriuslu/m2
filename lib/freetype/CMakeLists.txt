include(ExternalProject)
ExternalProject_Add(
    libfreetype
    URL ${CMAKE_CURRENT_LIST_DIR}/freetype-2.12.1.tar.gz
    URL_HASH SHA256=efe71fd4b8246f1b0b1b9bfca13cfff1c9ad85930340c27df469733bbb620938
    CMAKE_ARGS -DCMAKE_OSX_DEPLOYMENT_TARGET=${CMAKE_OSX_DEPLOYMENT_TARGET} -DCMAKE_INSTALL_PREFIX=${CMAKE_CURRENT_BINARY_DIR} -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} -DDISABLE_FORCE_DEBUG_POSTFIX=ON -DFT_DISABLE_ZLIB=ON -DFT_DISABLE_BZIP2=ON -DFT_DISABLE_PNG=ON -DFT_DISABLE_HARFBUZZ=ON -DFT_DISABLE_BROTLI=ON
)
