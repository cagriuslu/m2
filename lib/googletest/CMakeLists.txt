if (WIN32)
    add_library(googlemock SHARED IMPORTED GLOBAL)
    set_target_properties(googlemock PROPERTIES
            IMPORTED_LOCATION "${CMAKE_CURRENT_LIST_DIR}/googletest-release-1.12.1/build/install/bin/gmock.dll"
            IMPORTED_IMPLIB "${CMAKE_CURRENT_LIST_DIR}/googletest-release-1.12.1/build/install/lib/gmock.lib"
            INTERFACE_INCLUDE_DIRECTORIES "${CMAKE_CURRENT_LIST_DIR}/googletest-release-1.12.1/build/install/include"
            )
    add_library(googlemock_main SHARED IMPORTED GLOBAL)
    set_target_properties(googlemock_main PROPERTIES
            IMPORTED_LOCATION "${CMAKE_CURRENT_LIST_DIR}/googletest-release-1.12.1/build/install/bin/gmock_main.dll"
            IMPORTED_IMPLIB "${CMAKE_CURRENT_LIST_DIR}/googletest-release-1.12.1/build/install/lib/gmock_main.lib"
            INTERFACE_INCLUDE_DIRECTORIES "${CMAKE_CURRENT_LIST_DIR}/googletest-release-1.12.1/build/install/include"
            )
    add_library(googletest SHARED IMPORTED GLOBAL)
    set_target_properties(googletest PROPERTIES
            IMPORTED_LOCATION "${CMAKE_CURRENT_LIST_DIR}/googletest-release-1.12.1/build/install/bin/gtest.dll"
            IMPORTED_IMPLIB "${CMAKE_CURRENT_LIST_DIR}/googletest-release-1.12.1/build/install/lib/gtest.lib"
            INTERFACE_INCLUDE_DIRECTORIES "${CMAKE_CURRENT_LIST_DIR}/googletest-release-1.12.1/build/install/include"
            )
    add_library(googletest_main SHARED IMPORTED GLOBAL)
    set_target_properties(googletest_main PROPERTIES
            IMPORTED_LOCATION "${CMAKE_CURRENT_LIST_DIR}/googletest-release-1.12.1/build/install/bin/gtest_main.dll"
            IMPORTED_IMPLIB "${CMAKE_CURRENT_LIST_DIR}/googletest-release-1.12.1/build/install/lib/gtest_main.lib"
            INTERFACE_INCLUDE_DIRECTORIES "${CMAKE_CURRENT_LIST_DIR}/googletest-release-1.12.1/build/install/include"
            )
    target_link_libraries(googletest_main INTERFACE "googletest;googlemock_main;googlemock")
else()

endif()
