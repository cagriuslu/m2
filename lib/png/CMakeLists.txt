# Point to the install location of z
list(APPEND CMAKE_PREFIX_PATH "${CMAKE_CURRENT_BINARY_DIR}/../z")

include(ExternalProject)
ExternalProject_Add(
    libpng
    URL ${CMAKE_CURRENT_LIST_DIR}/libpng-1.6.47.tar.gz
    URL_HASH SHA256=084115c62fe023e3d88cd78764a4d8e89763985ee4b4a085825f7a00d85eafbb
    CMAKE_ARGS -DCMAKE_OSX_DEPLOYMENT_TARGET=${CMAKE_OSX_DEPLOYMENT_TARGET} -DCMAKE_PREFIX_PATH=${CMAKE_PREFIX_PATH} -DCMAKE_INSTALL_PREFIX=${CMAKE_CURRENT_BINARY_DIR} -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} -DPNG_SHARED=OFF
	DEPENDS libz
)
