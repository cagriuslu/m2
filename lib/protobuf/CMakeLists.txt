include(ExternalProject)

ExternalProject_Add(
        protobuf
        URL ${CMAKE_CURRENT_LIST_DIR}/protobuf-cpp-3.21.5.zip
        URL_HASH SHA256=5160E1EA2485CBA6329D1350D560D3DBB9246ABA91A46D076DDD971687DF253F
        CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${CMAKE_CURRENT_BINARY_DIR} -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} -Dprotobuf_BUILD_SHARED_LIBS=ON -Dprotobuf_BUILD_TESTS=OFF -Dprotobuf_WITH_ZLIB=OFF
)
