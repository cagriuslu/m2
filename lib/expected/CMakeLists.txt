include(FetchContent)
FetchContent_Declare(
    libexpected
    URL ${CMAKE_CURRENT_LIST_DIR}/expected-1.1.0.tar.gz
    URL_HASH SHA256=1db357f46dd2b24447156aaf970c4c40a793ef12a8a9c2ad9e096d9801368df6
)
set(EXPECTED_BUILD_PACKAGE OFF CACHE INTERNAL "Disable expected packages")
set(EXPECTED_BUILD_TESTS OFF CACHE INTERNAL "Disable expected testing")
